fgspectra/models.py

```diff
@@ -1,89 +1,134 @@
-# -*- coding: utf-8 -*-
 r"""
-Foreground models.
+Models of cross-spectra
 
 This module draws inspiration from FGBuster (Davide Poletti and Josquin Errard)
 and BeFoRe (David Alonso and Ben Thorne).
 """
-import fgspectra.frequency, fgspectra.power
+from abc import ABC, abstractmethod
 import numpy as np
+from . import frequency as fgf
+from . import power as fgp
 
 
-class CompositeModel:
-    def __init__(self):
-        self.params = {}
-        return
+class CrossSpectrum(ABC):
+    """Base class for cross-spectra."""
 
-    def model(self, nu_i, nu_j, ell, **kwargs):
-        """Compute the model at frequency and ell combinations.
+    @abstractmethod
+    def __call__(self, *args):
+        """Make it callable."""
+        pass
+
+
+class FactorizedCrossSpectrum(CrossSpectrum):
+    r"""Factorized cross-spectrum
+
+    Cross-spectrum of **one** component for which the scaling in frequency
+    and in multipoles are factorizable
+
+    .. math:: xC_{\ell}^{(ij)} = f(\nu_j) f(\nu_i) C_{\ell}
 
-        Specific models will override this method.
+    Parameters
+    ----------
+    sed : callable
+        :math:`f(\nu)`. It returns an array with shape ``(..., freq)``.
+        It can be :class:`fgspectra.frequency.SED`.
+    cl_args : callable
+        :math:`C_\ell`. It returns an array with shape ``(..., ell)``.
+        It can be :class:`fgspectra.power.PowerSpectrum`
+
+    Note
+    ----
+    The two (optional) sets of extra dimensions ``...`` must be
+    broadcast-compatible.
+    """
+
+    def __init__(self, sed, cl):
+        self._sed = sed
+        self._cl = cl
+
+    def __call__(self, sed_args, cl_args):
+        """Compute the model at frequency and ell combinations.
 
         Parameters
         ----------
-        nu_i : float
```

Comment:
Note that I removed the possibility of computing specific crosses: you can only compute all the crosses between an array of frequencies and select the entry you are interested in from the output array. I prefer, if anybody doesn't please chime in

Reply:
I would definitely like the ability to compute specific crosses.